### PR TITLE
Expand lint docs

### PR DESCRIPTION
**Pull request by chrisroberts**

### CODE REVIEW

File: docs/lint.md [hunk 1]
```diff
@@ -1,25 +1,113 @@
...
+* `Sfn::Lint::Definition`
+
+### Composition
+
+#### Long Form
+
+##### `Sfn::Lint::Definition`
+
+Definitions define a search expression to be applied to a given template. The search
+expression is a JMESPath compatible query expression. The matches
```
Comment (cwjohnston) [inline docs/lint.md]: Is there documentation for JMESPath query expression suitable for linking here?

Comment (chrisroberts) [inline docs/lint.md]: Yes!

File: docs/command-config.md [hunk 2]
```diff
@@ -887,6 +889,93 @@ $ sfn inspect
...
+| `--ignore-parameters` | Description | Parameters to ignore during modifications |
+| | Valid | `String` |
+| | Default | |
+| `--interactive-parameters` | Description | Prompt for template parameters |
+| | Valid | `TrueClass`, `FalseClass` |
+| | Default | true|
+| `--lint-directory` | Description | Directory containing lint rule sets |
+| | Valid | `String` |
+| | Default | |
+| `--local-rule-sets-only` | Description | Only apply rule sets provided by lint directory |
```
Comment (cwjohnston) [inline docs/command-config.md]: Seems like --local-rule-sets-only should have a default value here

**Comment by cwjohnston** [comment]
Aside from the comments I've added above, I think these docs are ready to ship! 👍 :shipit: 🏆

**Pull request merged**